cclient: implementing find_transactions API.

oopsmonk opened the pull request:
Fixes https://github.com/iotaledger/entangled/issues/44

Adding find_transactions implementation and test case.

# Test Plan:
* CI pass
* Local test
```
iota_http_service_t serv;
serv.host = "127.0.0.1";
serv.port = 14265;
serv.content_type = "application/json";
serv.version = 1;

init_json_serializer(&serv.serializer);

find_transactions_req_t *find_tran = find_transactions_req_new();
find_transactions_req_add_address(find_tran, "RVORZ9SIIP9RCYMREUIXXVPQIPHVCNPQ9HZWYKFWYWZRE9JQKG9REPKIASHUUECPSQO9JT9XNMVKWYGVAZETAIRPTM");
find_transactions_req_add_address(find_tran, "99999999IP9RCYMREUIXXVPQIPHVCNPQ9HZWYKFWYWZRE9JQKG9REPKIASHUUECPSQO9JT9XNMVKWYGVAZETAIRPTM");

find_transactions_res_t *find_tran_res = find_transactions_res_new();
iota_api_find_transactions(&serv, find_tran, find_tran_res);

// getting results
printf("address num = %d\n", find_transactions_res_hash_num(find_tran_res));
printf("0: %s\n", find_transactions_res_hash_at(find_tran_res, 0));
printf("1: %s\n", find_transactions_res_hash_at(find_tran_res, 1));

find_transactions_req_free(find_tran);
find_transactions_res_free(find_tran_res);
```

Commit:
cclient: implementing find_transactions API.

Adding find_transactions implementation and tests.

tsvisabo commented on common/errors.h:
```diff
@@ -156,6 +157,12 @@ enum retcode_t {
+
+  // Cclient Module
+  RC_CCLIENT_JSON_CREATE = 0x01 | RC_MODULE_CCLIENT | RC_SEVERITY_FATAL,
+  RC_CCLIENT_JSON_PARSE = 0x02 | RC_MODULE_CCLIENT | RC_SEVERITY_FATAL,
```
Should `RC_CCLIENT_JSON_PARSE` be Fatal?

tsvisabo commented on cclient/types/types.h:
```diff
@@ -8,10 +8,16 @@
 #endif
 
+#undef oom
+#define oom() printf("cclient oom\n")
```
you should use our logger `utils/logger/helper`

tsvisabo commented on cclient/service.h:
```diff
@@ -26,8 +27,8 @@ void init_iota_http_service(iota_http_service_t*, const char* const host,
 typedef struct {
-  size_t is_error;
   size_t duration;
+  retcode_t is_error;
```
prefix `is` should only be for bool types, `error` is good enough :)

tsvisabo commented on cclient/serialization/json/json_serializer.c:
```diff
@@ -8,13 +8,113 @@ void init_json_serializer(serializer_t* serializer) {
+  }
+
+  if (utarray_len(obj->approvees) > 0) {
+    cJSON* approvees = cJSON_CreateArray();
```
Please define all the local variables at the beginning of the function

tsvisabo commented on cclient/serialization/json/json_serializer.c:
```diff
@@ -8,13 +8,113 @@ void init_json_serializer(serializer_t* serializer) {
+    }
+  }
+
+  if (utarray_len(obj->tags) > 0) {
```
You should define a function that take `utarray` , json_root and label and return `cJSONArray` to avoid duplications

oopsmonk replied:
sure

Commit:
cclient: add utarray_to_json_array

add .vscode to .gitignore
rename iota_api_result_t.is_error to iota_api_result_t.error
